db/compaction/compaction_iterator_test.cc

```diff
@@ -184,7 +184,7 @@ class TestSnapshotChecker : public SnapshotChecker {
  public:
   explicit TestSnapshotChecker(
       SequenceNumber last_committed_sequence,
-      const std::unordered_map<SequenceNumber, SequenceNumber>& snapshots = {})
+      const std::unordered_map<SequenceNumber, SequenceNumber>& snapshots = {{}})
```

Comment:
please move this change to a separated commit, and explain the rationale behind this change.

Reply:
OK, I will separate them into two pr. Plz refer to the issue https://github.com/facebook/rocksdb/issues/3522, it may not friendly for g++4.9.

Reply:
i think the existing code is compliant to C++11. and why would you want to compile compile the test with [4 years old compiler](https://gcc.gnu.org/gcc-4.9/)?

Reply:
Sigh, it seems a sad history problem. If you would not like to make a compatibility, it's okay to make a local fit for me.

Reply:
i think it's the maintainers' call to include your fix or not. i am just wondering why. the reason i was curious is that, normally, the reason why we need to stick to an older version of compiler is just we have to compile our executables/libraries for supported older platforms/distros. but the tests is not in this scope.

Reply:
Briefly speaking, almost all projects/libs in our company are complied with g++4.9.2. As for me, I have to maintain compatible api lib version under g++4.9.2, 5.4.0, and 6.3.0, which also troubles me a lot. But this is life anyway. ;)

Reply:
but `compaction_iterator_test.cc` is not included by rocksdb library. so even if it does not compile, you can still compile and link against rocksdb using GCC 4.9.2 .

Reply:
Since we want to make use of built-in `db_bench_tool` as data generator, with the option `WITH_TESTS=ON`. But as you see, `${TESTS}` sources like `compaction_iterator_test` is involved.

Reply:
https://stackoverflow.com/questions/26947704/implicit-conversion-failure-from-initializer-list

Reply:
@pdillinger thanks for the link! so it's C++14 compliant.